pvlib/singlediode.py

```diff
@@ -794,6 +794,13 @@ def _lambertw(photocurrent, saturation_current, resistance_series,
     # Compute open circuit voltage
     v_oc = _lambertw_v_from_i(0., **params)
 
+    # Set elements <0 or close to 0 in v_oc to 0
+    if isinstance(v_oc, np.ndarray):
+        v_oc[(v_oc < 0) | ((v_oc > 0) & (v_oc < 1e-12))] = 0.
+    elif isinstance(v_oc, (float, int)):
+        if v_oc < 0 or 0 < v_oc < 1e-12:
```

Comment:
What I had in mind, is only setting small negative values to zero. Maybe it's caution for its own sake, but I'd prefer to keep larger negatives which must be from some user error or code problem other than roundoff.

```
v_oc[(v_oc <0) & (v_oc > -1e-12)] = 0.
```